src/formatter/helpers/summary_helpers.js

```diff
@@ -8,6 +8,7 @@ const STATUS_REPORT_ORDER = [
   Status.UNDEFINED,
   Status.PENDING,
   Status.SKIPPED,
+  Status.RETRIED,
```

Comment:
Giving the status `RETRIED` doesn't tell me whether the step/scenario passed or failed. This doesn't seem like a useful status to me.

IIRC [cucumber-ruby](https://github.com/cucumber/cucumber-ruby/) uses a `FLAKY` status to indicate that the result was initially `FAILED`, then `PASSED` in a retry. If all the retries are `FAILED` the final result is also `FAILED`.

Cucumber implementations should be consistent in their behaviour and reporting. Could you change this to be consistent with cucumber-ruby please?

Reply:
@aslakhellesoy thank you for your time
I've created a new status `FLAKY` to do what happens in cucumber-ruby (looked at the tests there) - see the second commit
the `RETRIED` is still there because it is used to report how often the step was retried

Reply:
Thanks @individual-it for the merging and fixes.
@aslakhellesoy there was a quite long discussion with @charlierudolph in regards to FLAKY vs RETRIED. 
You can find the discussion on the old PR starting at the comment from on 21 Dec 2018 :)

Reply:
here the link to the discussion: https://github.com/cucumber/cucumber-js/pull/1114#pullrequestreview-187261764
Which way to go? Should I revert it back to be status "RETRIED" (delete last commit)? Or can we merge it as it is? should we have a switch that does the one or the other?

Reply:
I definitely prefer RETRIED (you can see the linked discussion for my reasons). I don't like the idea of having a switch

Reply:
To be extra clear I think we could do "FAILED_AND_RETRIED" for a pickle that fails and is then retried.

Reply:
@charlierudolph there basically two new states. first "FAILED_AND_RETRIED_AND_STILL_FAILS" what currently ends up as "FAILED"
and then there is the status "FAILED_AND_RETRIED_AND_NOW_WORKS" that is what is currently "FLAKY"

I see two questions:
1. is just the naming of the status RETRIED vs FLAKY, naming is a one of the two hard problems in computing
2. what to report
the original PR would only report in in the header
```
      1) Scenario: Flaky (attempt 2) # features/a.feature:2
      ✔ Given a flaky step # features/step_definitions/cucumber_steps.js:5
      1 scenario (1 passed)
      1 step (1 passed)
```

after the change to flaky status it reports more
```
      1) Scenario: Flaky (attempt 2) # features/a.feature:2
      ✔ Given a flaky step # features/step_definitions/cucumber_steps.js:5
      1 scenario (1 flaky)
      1 step (1 passed)
```

How are this decisions made in this project? I personally don't care too much ether way, just care about get this function merged ;-)

Reply:
The main difference to me is what the log of events look like. Cucumber streams events as they happen and I want that to be as clear as possible.

1) My idea was that for a test case that fails and then retries, the status would be 1) RETRIED, 2) PASSED or FAILED instead of 1) FAILED,  2) FLAKEY or FAILED. The main difference being what info you when the first failure happens about whether or not it will be retried.

I guess I'm fine moving forward with current solution if we also add a "retried: true" to the test result for a pickle that will be retried so we have that information.

As to how the decisions are made, anyone can offer solutions and ideally we debate them until we agree on a solution. The maintainers are the ultimate decision makers since they can merge code and they are who have to deal with maintaining the feature (and making sure everything else continues to work). I'm the main maintainer for cucumber-js while @aslakhellesoy is the creator of cucumber / maintainer of all variants.

Reply:
@charlierudolph I've added a `retried:true` to the result, what goes along with the FLAKY status see 1a9f98cf3c7c117d0258048b5c9fc57b9eb81d4c
is that what you mean by:
> I guess I'm fine moving forward with current solution if we also add a "retried: true" to the test result for a pickle that will be retried so we have that information.

Reply:
IIRC the FLAKEY result status from the Ruby Cucumber is in a summary for the end of the run, not info that you'd see about an individual pickle execution result. If we're talking about a stream of events during the run, I think having straightforward "FAILED" then "PASSED" results from the runner is OK, as some higher-level thing can make sense of those, see they're for the same test case and work out that it have been retried. Still a `retried: true` flag can't hurt as extra confirmatory information, and I think this is a good compromise for now to move us forward.

In the future, we could also have another event entirely in the stream notifying about the decision to retry a test case.

Reply:
Coming back with what I updated the formatters to be. The scenario is reported as passing but the failed attempt is reported as a warning. See features/retry.feature for example